coordinator/src/services/validation/util.ts

```diff
@@ -0,0 +1,14 @@
+import {isDeepStrictEqual} from "util"
+
+export function getUniqueValues<T>(allValues: Array<T>): Array<T> {
+  return allValues.reduce(
+    (existingUniqueValues, valueToTest) => {
+      if (existingUniqueValues.every((uniqueValue) => !isDeepStrictEqual(uniqueValue, valueToTest))) {
+        return existingUniqueValues.concat(valueToTest)
+      } else {
+        return existingUniqueValues
+      }
+    },
+    [allValues.shift()]
```

Comment:
Presumably, the reason we're popping the first value off the array here is because we need something to seed the reduce function with?

Reply:
Yeah that's it